File: src/sugtao4423/lod/dialog_onClick/Dialog_regButtonClick.java
package sugtao4423.lod.dialog_onClick;

import android.text.Editable;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.EditText;

public class Dialog_regButtonClick implements OnClickListener{

	private EditText regEdit;
	private String regText;

	public Dialog_regButtonClick(EditText regEdit, String regText){
		this.regEdit = regEdit;
		this.regText = regText;
	}

	@Override
	public void onClick(View v){
		int start = regEdit.getSelectionStart();
		int end = regEdit.getSelectionEnd();
		Editable editable = regEdit.getText();
		editable.replace(Math.min(start, end), Math.max(start, end), regText);
	}
}

File: app/src/main/java/sugtao4423/lod/dialog_onclick/Dialog_regButtonClick.kt
package sugtao4423.lod.dialog_onclick

import android.view.View
import android.widget.Button
import android.widget.EditText
import kotlin.math.max
import kotlin.math.min

class Dialog_regButtonClick(private val regEdit: EditText) : View.OnClickListener {

    override fun onClick(v: View?) {
        val start = regEdit.selectionStart
        val end = regEdit.selectionEnd
        regEdit.text.replace(min(start, end), max(start, end), (v as Button).text)
    }

}